### ISSUE TITLE
Usage for multimodal data (data with fundamentally different characteristics)

### ISSUE THREAD

**Issue by trangham283** [shap/shap#2130]
Hello, thank you for a great library. I was wondering if it is possible (or reasonable) to use SHAP for explaining models with mixed (discrete and continuous) inputs. Specifically, my model takes as input a list of tokens (sentence) and a matrix of acoustic features corresponding to that utterance.

For the text input, I'm following the example for emotion classification. Specifically:

```
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
from transformers import AutoTokenizer, AutoModelForSequenceClassification
import shap
import scipy as sp
from datasets import load_dataset
import torch
from torch import nn

tokenizer = AutoTokenizer.from_pretrained("nateraw/bert-base-uncased-emotion",use_fast=True)
text_model = AutoModelForSequenceClassification.from_pretrained("nateraw/bert-base-uncased-emotion")

dataset  = load_dataset("emotion", split = "train")
data = pd.DataFrame({'text':dataset['text'],'emotion':dataset['label']})
id2label = text_model.config.id2label
label2id = text_model.config.label2id
labels = sorted(label2id, key=label2id.get)
```

I have a dummy model that projects acoustic features to the number of classes:

```
class_num = 6
speech_dim = 100
sp_model = nn.Linear(speech_dim, class_num)      
```

The overall model is defined as follows; iin this case, I'd also like to use another text-based feature like previous sentence as context:

```
class dummyModel(nn.Module):
    def __init__(self, text_model, speech_model=None):
        super(dummyModel, self).__init__()
        self.text_model = text_model
        self.speech_model = speech_model
        
    def forward(self, tvx, tvbg, tvs):
        # tvx: current utterance to classify; tvbg: context utterance; tvs: speech features
        attn_x = (tvx != 0).type(torch.int64)
        attn_bg = (tvbg != 0).type(torch.int64)

        o1 = self.text_model(tvx, attention_mask=attn_x)[0]
        o2 = self.text_model(tvbg, attention_mask=attn_bg)[0]
        outcomb = o1 + o2
        if self.speech_model is not None:
            o_speech = self.speech_model(tvs)
            outcomb = outcomb + o_speech
        return outcomb
```

Toy data & model:

```
x_text = data['text'][0:10]
bg_text = data['text'][10:20]
tvs = torch.rand((10, speech_dim))

tvx = torch.tensor([tokenizer.encode(v, padding='max_length', max_length=128,truncation=True) for v in x_text])
tvbg = torch.tensor([tokenizer.encode(v, padding='max_length', max_length=128,truncation=True) for v in bg_text])

attn_x = (tvx != 0).type(torch.int64)
attn_bg = (tvbg != 0).type(torch.int64)

batch = [tvx, tvbg, tvs]

my_model = dummyModel(text_model)
```

Ideally, I'd like to get an explainer: 

```
e = shap.Explainer(my_model, ...)
```

However, from my understanding, explaining text inputs at the token level uses the PartitionExplainer, while a use case like this (multiple inputs) maybe GradientExplainer is more appropriate ([from this example](https://shap-lrjball.readthedocs.io/en/latest/example_notebooks/gradient_explainer/Multi-input%20Gradient%20Explainer%20MNIST%20Example.html)).

I think the issue is that the tokens are masked using the tokenizer, where essentially sampling of missingness is based on using the [MASK] token (and in the embedding space the whole vector is masked). On the other hand, to explain other (e.g. acoustic features), each dimension is considered independently. 

Is there a way to get around this? Any pointers would be appreciated, thank you very much!
